app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala

```diff
@@ -31,9 +31,45 @@ import javax.inject.Inject
 
 class MessagesController @Inject() (cc: ControllerComponents)(implicit val materializer: Materializer) extends BackendController(cc) with StreamingParsers {
 
-  def post(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+  def channelResponseGB(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
     request: Request[Source[ByteString, _]] =>
       request.body.runWith(Sink.ignore)
-      Accepted
+
+      validateHeaders(request.headers, "gb") match {
+        case true => Ok
+        case _    => BadRequest
+      }
+  }
+
+  def channelResponseXI(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+    request: Request[Source[ByteString, _]] =>
+      request.body.runWith(Sink.ignore)
+
+      validateHeaders(request.headers, "xi") match {
+        case true => Ok
+        case _    => BadRequest
+      }
   }
+
+  def validateHeaders(headers: Headers, officeCode: String): Boolean = {
+    var result = isHeaderExists("content-type", headers) &&
+      isHeaderExists("accept", headers) &&
+      headers.get("authorization").nonEmpty &&
+      headers.get("x-message-type").nonEmpty &&
+      headers.get("x-correlation-id").nonEmpty &&
+      headers.get("date").nonEmpty
```

Comment:
Looks like we're missing the check for x-request-id

Reply:
I skipped this on purpose, as it's Optional as per the spec